packages/build/tests/plugins/constants/fixtures/build_default/plugin.js

```diff
@@ -1,6 +1,6 @@
 module.exports = {
   name: 'netlify-plugin-test',
   onInit({ constants: { BUILD_DIR } }) {
-    console.log(BUILD_DIR)
+    console.log(BUILD_DIR, BUILD_DIR === __dirname)
```

Comment:
Update the test to ensure users with no `build.publish` gets a publish directory that is the package root.